frame/support/src/weights/weight_v2.rs

```diff
@@ -0,0 +1,454 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use codec::{Decode, Encode, MaxEncodedLen};
+use core::ops::{Add, AddAssign, Div, Mul, Sub, SubAssign};
+use sp_runtime::{
+	traits::{Bounded, CheckedAdd, CheckedSub, One, Zero},
+	Perquintill, RuntimeDebug,
+};
+
+use super::*;
+
+pub type RefTimeWeight = u64;
```

Comment:
Is the plan to eventually make this type private? Some docs would be nice here.

Reply:
no, i think it will stick around. i added docs

Reply:
I am kind of allergic to primitive type aliases. I think we should either use the primitive type or crate a newtype `struct RefTimeWeight(u64)` with a `Deref` impl on it. Primitive type aliases are just the worst of both worlds: Adding cognitive load by making you look up another time but offer nothing in refactoring advantages as a newtype would do.

Reply:
okay, i am happy enough to remove it in the next round

Reply:
Maybe a better way is to use a wrapper struct like @athei mentioned, but implement all the basic math traits on it like `Add`, `Sub`, `Mul` and `Div`, so that `RefTimeWeight` can act almost like a number.

Reply:
i dont think we need to do that, u64 should be fine for a single field here.

The issue is only when the whole struct is represented as u64.

Reply:
> Type aliases are used are used to reduce cognitive load, not to increase it, imo

The road to hell is paved with good intentions. I never felt they helped me. They just hide the truth from me and offer zero type safety. But I am not gonna die on that hill.